### ISSUE TITLE
Import trajectories from PDB files

### ISSUE THREAD

**Issue by joerivan** [OpenChemistry/avogadrolibs#1524]
It could be a good add if Avogadro could import trajectories from PDB files. The old Avogadro had this feature, but it does not seem present/working in Avogadro 2. 

Similar to loading a 'normal' PDB file, the feature should allow the user the drag and drop a PDB file into Avogadro, and Avogadro should read the PDB file, where each geometry that is separated by an `ENDMDL`[^1] should be imported as a separate frame.

 Switching between the frames should then be possible (as normal) via the `Animation Tool`. 

An alternative would be to import with a separate tool/menu. For example, adding an option to import trajectories inside of the `Animation Tool` (specific for PDB files). I believe that the first suggestion would be more user-friendly.

[^1]:In PDB files coordinates are read until either the file ends or the keyword `ENDMDL` is encountered. Using this, a PDB file can be used as a trajectory format. The different steps are then separated by an `ENDMDL`.

**Comment by ghutchis** [comment]
I'm going to mark this as "help wanted" and "good first issue" since I've got a few other things on my plate right now.

The code for this is in https://github.com/OpenChemistry/avogadrolibs/blob/bb080b5bfd45b8c197702d802888687450c6f5d5/avogadro/io/pdbformat.cpp#L59

An example of adding frames is in the XYZ code: https://github.com/OpenChemistry/avogadrolibs/blob/bb080b5bfd45b8c197702d802888687450c6f5d5/avogadro/io/xyzformat.cpp#L137

**Comment by vaibhav-1508** [comment]
Hey there @ghutchis , I was interested in working on this issue but had problem in figuring out how to set up the repository locally. Apart from forking and cloning, what next do I have to do?. Are there any docs which I can refer to?

**Comment by ghutchis** [comment]
The README points to https://two.avogadro.cc/install/build.html

**Comment by Himanshuu23** [comment]
> It could be a good add if Avogadro could import trajectories from PDB files. The old Avogadro had this feature, but it does not seem present/working in Avogadro 2.
> 
> Similar to loading a 'normal' PDB file, the feature should allow the user the drag and drop a PDB file into Avogadro, and Avogadro should read the PDB file, where each geometry that is separated by an `ENDMDL`[1](#user-content-fn-1-c00b3e6e91578adc4905e4acbfe308a3) should be imported as a separate frame.
> 
> Switching between the frames should then be possible (as normal) via the `Animation Tool`.
> 
> An alternative would be to import with a separate tool/menu. For example, adding an option to import trajectories inside of the `Animation Tool` (specific for PDB files). I believe that the first suggestion would be more user-friendly.
> 
> ## Footnotes
> 1. In PDB files coordinates are read until either the file ends or the keyword `ENDMDL` is encountered. Using this, a PDB file can be used as a trajectory format. The different steps are then separated by an `ENDMDL`. [↩](#user-content-fnref-1-c00b3e6e91578adc4905e4acbfe308a3)

@joerivan Please assign me to work on this issue. I have solved this issue 95% just I'm trying to figure out one part but would solve it ASAP.

**Comment by ghutchis** [comment]
We generally don't "assign" issues except to mark that someone from the core development team is working on it.

If you want to find the correct files, you need to look at:
https://github.com/OpenChemistry/avogadrolibs/blob/master/avogadro/io/pdbformat.cpp

An example of reading multiple coordinate sets can be found in `xyzformat.cpp`, e.g.
https://github.com/OpenChemistry/avogadrolibs/blob/dfbd6f6bf9fefc4a13278a9c2fefd26f2dc1b98a/avogadro/io/xyzformat.cpp#L135

Basically, you push back each frame:
```c++
Array<Vector3> positions;
positions.reserve(numAtoms);
unsigned coordSet = 0;
mol.setCoordinate3d(positions, 0);
```

**Comment by Himanshuu23** [comment]
Raised an PR for it.
